Fixed compatibility with Posgresql, interited AutoField, closed #90

hynekcer opened the pull request:
I think that it also simplifies the future development.

philchristensen commented:
@hynekcer: Seems like we should do a point-release for something like this. Anything on the horizon I should wait for?

Incidentally, I'll soon be back to using django-salesforce again fairly heavily at my main job. I'm encouraging the sales/marketing deparments to convert fully to Salesforce from a home-grown CRM specifically so I can have a standard interface to interact with their data ;-)

hynekcer commented:
I have currently 24 commits not added to any pull request, waiting for final testing, and 10 commits waiting in the queue for finalizing. The most important open feature is the fix for ForeignKey filter (issue 43) from the end of October. It was relative easy but it is complicated to write tests

I'm reoganizing the majority of commits listed in the release draft. Sometimes is the order important for better testing, sometimes I want to do folow-up steps of one feature in a successive group of commits even though I developed them like multiplexed by othe features, sometimes I changed anything in the example model and tests must be updated. Once I found a bug much later that a test did not tested anything. I try to follow the standard that every commit in PR works as expected or a known bug is described in the commit message that will be fixed exactly in the next commit, if it is better to separate them

E.g. the SFDC API 32.0 caused a bug, because they added an incorrect ForeignKey that is not related to any model (undocumented and probably also unused). So I first demonstrated the bug by increasing the API, then I fixed inspectdb and finally I fixed also the guaranted validity of the exported Django model. (by automatic renaming such ForeignKey to TextField and writing a big comment behind it)

philchristensen commented:
So you found a bug in the Salesforce v32.0 API because of the django-salesforce unit tests? That's great!

I'm glad to do a release whenever. At this point it seems like mostly edge cases that have been fixed since the last release.

-phil

> On Jan 25, 2015, at 11:09 AM, Hynek Cernoch <notifications@github.com> wrote:
> 
> I have currently 24 commits not added to any pull request, waiting for final testing, and 10 commits waiting in the queue for finalizing. The most important open feature is the fix for ForeignKey filter (issue 43) from the end of October. It was relative easy but it is complicated to write tests
> 
> I'm reoganizing the majority of commits listed in the release draft. Sometimes is the order important for better testing, sometimes I want to do folow-up steps of one feature in a successive group of commits even though I developed them like multiplexed by othe features, sometimes I changed anything in the example model and tests must be updated. Once I found a bug much later that a test did not tested anything. I try to follow the standard that every commit in PR works as expected or a known bug is described in the commit message that will be fixed exactly in the next commit, if it is better to separate them
> 
> E.g. the SFDC API 32.0 caused a bug, because they added an incorrect ForeignKey that is not related to any model (undocumented and probably also unused). So I first demonstrated the bug by increasing the API, then I fixed inspectdb and finally I fixed also the guaranted validity of the exported Django model. (by automatic renaming such ForeignKey to TextField and writing a big comment behind it)
> 
> —
> Reply to this email directly or view it on GitHub.
>
